src/error.rs

```diff
@@ -23,6 +24,12 @@ pub enum Error {
     VarDecode(&'static str),
     #[error("{0}")]
     TableRead(String),
+    #[error("Database Closed")]
+    DBClosed,
+    #[error("Lock Poison")]
+    PoisonError(String),
```

Comment:
I suggest to panic. `Poison` is a hint of bug, we don't know what can go wrong when bug happen.